firmware/src/network_model.c

```diff
@@ -147,8 +155,8 @@ static int LookupIndex(unsigned int id) {
  *
  * This randomly fails and returns 0.0.0.0.
  */
-static uint32_t GetDHCPAddress() {
-  if (Random_PseudoGet() % DHCP_FAILURE_RATIO == 0) {
+static uint32_t GetDHCPAddress(bool can_fail) {
+  if (can_fail && Random_PseudoGet() % DHCP_FAILURE_RATIO == 0) {
```

Comment:
Presumably flakeyness isn't a good thing, and testing flakeyness doesn't work too well either?